arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -121,12 +123,103 @@ def run_timerlat(
+                                cpu = int(cpu_val.match(line_list[i]).group(1))
+                                break
+                            except (ValueError, AttributeError):
+                                line_list.pop(i)
+                            except IndexError as error:
+                                print(
+                                    "Unable to determine CPU number; Skipping time "
+                                    f"series collection: {error}\n{line}"
+                                )
+                                timeseries_dict = {}
+                                break
+
+                    # Because the tracer format is dependent on the underlying OS and
+                    # cannot be controlled by the container, check the tracer output
+                    # format and break gracefully if we don't recognize it
+                    try:
+                        uptimestamp = float(line_list[3][:-1])
+                        timestamp = str(
+                            datetime.fromtimestamp(uptimestamp + uptime_offset)
+                            .astimezone()
+                            .isoformat()
+                        )
+                        context = str(line_list[6])
+                        latency = int(line_list[8])
+
+                    except (IndexError, ValueError) as error:
+                        print(
+                            "Unknown tracer format; Skipping time series collection: "
+                            f"{error}\n{line}"
+                        )
+                        timeseries_dict = {}
+                        break
+
+                    # The trace collects for all CPUs, so skip any CPU we did not select
+                    # via the input to the plugin
+                    if params.cpus and int(cpu) not in params.cpus:
+                        continue
+                    # Create a separate time series for each CPU + context combination
+                    cpu_context = f"cpu{cpu}_{context}"
+                    if cpu_context not in timeseries_dict:
+                        timeseries_dict[cpu_context] = []
+                    if cpu_context not in last_uptimestamp:
+                        last_uptimestamp[cpu_context] = 0
+                    if last_uptimestamp[cpu_context] and (
+                        uptimestamp - last_uptimestamp[cpu_context]
+                        < params.time_series_resolution
+                    ):
+                        continue
+                    last_uptimestamp[cpu_context] = uptimestamp
+                    timeseries_dict[cpu_context].append(
+                        latency_timeseries_schema.unserialize(
+                            {
+                                "timestamp": timestamp,
+                                "latency_ns": latency,
+                            }
+                        )
+                    )
```

Comment:
You're just taking occasional samples and pitching the rest of the high-res data?  You don't want to aggregate the samples from each interval somehow?

Reply:
I've been debating this. Since the existing data consists of point-in-time measurements, I'm not sure if it's better to simply reduce the resolution of that collection, as I have done here, or if I should take the mean of the "skipped" interim values and report that instead of the point value.

Reply:
Yup.  I just wanted to confirm.